src/Form/EmbedButtonForm.php

```diff
@@ -168,6 +168,39 @@ protected function copyFormValuesToEntity(EntityInterface $entity, array $form,
   }
 
   /**
+   * Builds a list of entity type labels suitable for embed button options.
+   *
+   * Configuration entity types without a view builder are filtered out while
+   * all other entity types are kept.
+   *
+   * @return array
+   *   An array of entity type labels, keyed by entity type name.
+   */
+  protected function getFilteredEntityTypes() {
+    $options = array();
+    $definitions = $this->entityManager->getDefinitions();
+
+    foreach ($definitions as $entity_type_id => $definition) {
+      // Don't include configuration entities which do not have a view builder.
+      if ($definition->getGroup() != 'configuration' || $definition->hasViewBuilderClass()) {
```

Comment:
@davereid I'm not sure how this is going to work since there'll be some foramtters whose `entity_types` value is FALSE and hence are applicable to all entity types. So, no matter which entity type we choose, they'll always be present. This again leaves us with two options:

- Use the same `hasViewBuilder` approach, or
- Check whether that entity type has any formatter besides default ones

What do you think?